lib/src/test/groovy/com/cloudbees/groovy/cps/CpsTransformerTest.groovy

```diff
@@ -877,4 +878,35 @@ def (c, d) = ['fourth']
 return [a, b, c, d].join(' ')
 ''') == "first second fourth null"
     }
-}
+
+    @Issue("JENKINS-47363")
+    @Test
+    void excessiveListElements() {
+        def l = 0..250
+        def s = l.join(",\n")
+        try {
+            assert evalCPS("""
+def b = [${s}]
+return b.size()
+""") == 251
+        } catch (Exception e) {
+            assert e instanceof MultipleCompilationErrorsException
```

Comment:
I guess this was some other error type previously?

Reply:
Correct.